shm/include/ship.hpp

```diff
@@ -0,0 +1,41 @@
+#pragma once
+
+#include <string>
+#include <vector>
+
+#include "cargo.hpp"
+#include <memory>
```

Comment:
Why are you doing memory inclusion in such weird way instead of putting it with the rest of standard library inclusion?

Reply:
Thanks for attention. Our project is working enviorment so sometimes you will see things that shouldn't  be here.